src/ethereum/EthereumNetwork.ts

```diff
@@ -515,6 +515,10 @@ export class EthereumNetwork {
       const apiKey = this.ethEngine.initOptions[cleanKey]
       if (typeof apiKey === 'string') {
         url = url.replace(match[0], apiKey)
+      } else if (apiKey == null) {
+        throw new Error(
+          `Missing ${cleanKey} in 'initOptions' for ${this.ethEngine.currencyInfo.pluginId}`
+        )
       } else {
         throw new Error('Incorrect apikey type for RPC')
```

Comment:
I looked at the type for `apiKey` and it looks like it could be an array of strings. Shouldn't we handle this case for `evmScanApiKey`? Maybe a simple random pick of one the keys

Reply:
No. Evmscan servers are not an RPC servers but a completely different "type". This PR is only for RPC servers and I don't want to scope creep the PR. That would also require another hour of testing.

Reply:
Ah, didn’t catch that difference. Thanks for correcting me here